src/main/java/com/mongodb/hibernate/internal/type/BsonTypeUtils.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.internal.type;
+
+import com.mongodb.hibernate.internal.FeatureNotSupportedException;
+import java.math.BigDecimal;
+import org.bson.BsonBinary;
+import org.bson.BsonBoolean;
+import org.bson.BsonDecimal128;
+import org.bson.BsonDouble;
+import org.bson.BsonInt32;
+import org.bson.BsonInt64;
+import org.bson.BsonNull;
+import org.bson.BsonObjectId;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+import org.bson.types.Decimal128;
+import org.bson.types.ObjectId;
+import org.jspecify.annotations.Nullable;
+
+public final class BsonTypeUtils {
+
+    private BsonTypeUtils() {}
+
+    public static BsonValue toBsonValue(@Nullable Object value) {
```

Comment:
Do we already know that this method will be used in more places than just `AbstractMqlTranslator`? If not, we can make it a `private static` method of `AbstractMqlTranslator`.

Reply:
In this PR it is not meant to be reused so I moved it as private static method in `AbstractMqlTranslator` as you suggested.